internal/pkg/mount/v3/manager.go

```diff
@@ -0,0 +1,236 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package mount
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/siderolabs/talos/internal/pkg/xfs"
+	"github.com/siderolabs/talos/internal/pkg/xfs/fsopen"
+)
+
+// Manager is the filesystem manager for mounting and unmounting filesystems.
+type Manager struct {
+	fs xfs.FS
+
+	target  string
+	printer func(string, ...any)
+
+	selinuxLabel          string
+	shared                bool
+	skipIfMounted         bool
+	keepOpen              bool
+	mountattr             int
+	extraDirs             []string
+	extraUnmountCallbacks []func(m *Manager)
+
+	point *Point
+}
+
+// NewManager creates a new Manager with the given options.
+func NewManager(opts ...ManagerOption) *Manager {
+	m := &Manager{}
+
+	for _, opt := range opts {
+		opt.set(m)
+	}
+
+	return m
+}
+
+// Mount mounts a filesystem with the given options.
+func (m *Manager) Mount() (*Point, error) {
+	printer := discard
+	if m.printer != nil {
+		printer = m.printer
+	}
+
+	for _, dir := range m.extraDirs {
+		printer("creating directory tree %q", dir)
+
+		if err := os.MkdirAll(dir, 0o755); err != nil {
```

Comment:
should we make sure the caller calls this code from `/`? otherwise the code could create a directory under `/foo` for example

Reply:
I mean if there is some code that does `os.Chddir` and calls this api doesn't `os.Mkdirall` creates under that folder 🤔

Reply:
I created check if the path is absolute, otherwise it will return error.